featuretools/primitives/options_utils.py

```diff
@@ -53,14 +56,17 @@ def generate_all_primitive_options(all_primitives,
     global_ignore_variables = ignore_variables.copy()
     # for now, only use primitive names as option keys
     for primitive in all_primitives:
-        if not isinstance(primitive, str):
-            primitive = primitive.name
-        if primitive in primitive_options:
+        if primitive in primitive_options and primitive.name in primitive_options:
+            msg = "Options present for primitive instance and generic " \
+                  "primitive class (%s), primitive instance will not use generic " \
+                  "options" % (primitive.name)
+            logger.warning(msg)
```

Comment:
I don't think we verify the warning message is logged as expected. Maybe we can update `test_primitive_options_instantiated_primitive` to check the warning is raised properly as well.